### ISSUE TITLE
about the implementation details in MadNet

### ISSUE THREAD

**Issue by uestc-cheng** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#61]
First I want to thank the authors for publishing the source code. Your work has addressed the key problem in applying deep learning based stereo matching algorithms in reality applications.
I have some minor questions about the implementation details:
1. In the "_make_disp" function in MadNet.py, why does the estimated disparity maps need to multiply a negative number "-20"?
2. To get variable "u5", why does the upsampled "V6" need to first multiply a postive number "20" and then divide the scale number, "scale[5]" for example?
3. In the "_stereo_cost_volume_correlation" function in MadNet.py, is it necessary to concatenate the "reference" features with the cost, which seems not memtioned in the paper? 
4. why not use batch norm in the "_pyramid_features" function in MadNet.py? I find the default setting of the "sharedLayers.conv2d" function declaration is "batch_norm=False". Is there any specific considerations?
5. In the "get_supervised_loss" fuction in "loss_factory”, I find the L1 loss is computed with all upsampled disparity maps with groundtruth maps, but in paper it said "compute the ... errors between disparity maps estimated at each resolution and downsampled groundtruth labels". It is a litter confusing. And does that affect training effect?
Finally I want say thanks for the authors efforts!

**Comment by AlessioTonioni** [comment]
Hello,
1. We train MadNet to predict disparity scaled by 20. It's a common trick in many disparity and flow estimation paper, empirically speaking it helps convergence.
2. It brings the predicted disparity to absolute scale and then scale it by scale[5]
3. yes, the code is the one used in the paper. Maybe we forgot to mention the concatenation there.
4. I thinkthat we tried to enable batchnorm but it was not helping, especially in the online adaptation part it was making things more complicated
5. If I remember well we predict scaled disparities, upsample them by knowing the scale and compute the loss at the higher scale. It's easier to balance training like this because all losses will be computed on the same number of pixels.

**Comment by uestc-cheng** [comment]
Thanks for your reply :）
